datafusion/optimizer/src/analyzer/subquery.rs

```diff
@@ -300,17 +300,16 @@ fn can_pullup_over_aggregation(expr: &Expr) -> bool {
     }) = expr
     {
         match (left.deref(), right.deref()) {
-            (Expr::Column(_), right) if right.to_columns().unwrap().is_empty() => true,
-            (left, Expr::Column(_)) if left.to_columns().unwrap().is_empty() => true,
+            (Expr::Column(_), right) if !right.any_column_refs() => true,
```

Comment:
Not related to this PR, but I wonder if we could simplify this code to 
```rust
            (Expr::Column(_), right) => !right.any_column_refs(),
            (left, Expr::Column(_)) => !left.any_column_refs(),
            (Expr::Cast(Cast { expr, .. }), right)
                if matches!(expr.deref(), Expr::Column(_)) => !right.any_column_refs(),
            (left, Expr::Cast(Cast { expr, .. }))
                if matches!(expr.deref(), Expr::Column(_)) => !left.any_column_refs(),
            (_, _) => false,
```

Reply:
That is an excellent idea -- done in cefb7bd0f